frontend/src/components/Switch.tsx

```diff
@@ -0,0 +1,45 @@
+import { useState } from "react";
+import { useColorMode } from "../providers/ColormodeProvider";
+
+function Switch() {
+  const [toggle, setToggle] = useState(true);
+  const toggleClass = " transform translate-x-5";
+  const { toggleTheme } = useColorMode();
+
+  const changeMode = async (mode: boolean) => {
+    setToggle(mode);
+    toggleTheme();
+  };
+
+  return (
+    <div className="w-14 h-8 mr-5">
+      <div className="flex flex-col">
+        {/*   Switch Container */}
+
+        <div
+          className="md:w-14 md:h-7 w-12 h-6 flex items-center bg-gray-400 rounded-full p-1 cursor-pointer"
+          onClick={() => {
+            changeMode(!toggle);
+          }}
+        >
+          <div
+            className={
+              "bg-black md:w-6 md:h-6 h-5 w-5 rounded-full shadow-md transform duration-300 ease-in-out" +
+              (toggle ? null : toggleClass)
+            }
+          >
```

Comment:
use the classnames lib, see usage in the Button component

```suggestion
          <div
            className={cn(
              "bg-black md:w-6 md:h-6 h-5 w-5 rounded-full shadow-md transform duration-300 ease-in-out",
              theme === 'dark' && 'translate-x-5'
            )}
          >
```

Reply:
I think you missed the `transition-transform` class here, does the animation work without it?